### PR TITLE
Implement inheritance of initializers with generic parameters [4.2]

### PR DESCRIPTION
**Pull request by slavapestov**
Fixes <https://bugs.swift.org/browse/SR-3848>.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test

File: lib/Sema/CodeSynthesis.cpp [hunk 1]
```diff
@@ -2005,6 +2006,109 @@ static void createStubBody(TypeChecker &tc, ConstructorDecl *ctor) {
...
+    auto source =
+      GenericSignatureBuilder::FloatingRequirementSource::forAbstract();
+    auto *superclassSig = superclassCtor->getGenericSignature();
+
+    unsigned superclassDepth = 0;
+    if (auto *genericSig = superclassDecl->getGenericSignature())
+      superclassDepth = genericSig->getGenericParams().back()->getDepth() + 1;
+
+    // We're going to be substituting the requirements of the base class
+    // initializer to form the requirements of the derived class initializer.
```
[the remark this answers is not in the document]
Comment (jrose-apple) [reply lib/Sema/CodeSynthesis.cpp]: Keeping it's probably the best option for now. Who knows what would happen if it was `required`?

**Pull request merged**